In [3]:
from utils.test_env import EnvTest

In [5]:
import itertools

In [6]:
env = EnvTest()

actions = list(range(5))
# construct all possible trajectories
trajs = list(itertools.product(*[actions] * 5))

rewards = []
for t in trajs:
    obs = env.reset()
    rew = 0
    done = False
    for a in t:
        obs_next, reward, done, info = env.step(a)
        rew += reward
    assert done is True
    rewards.append(rew)

# sort trajectories by their corresponding rewards
sorted_trajs = reversed(sorted(zip(rewards, trajs), key=lambda t: t[0]))

# print top 10
print('Reward\tTrajectory')
for k, (rew, traj) in enumerate(sorted_trajs):
    print('{0}\t{1}'.format(rew, traj))
    if k == 9:
        break

Reward	Trajectory
4.1	(4, 2, 1, 2, 1)
4.1	(2, 1, 2, 1, 0)
4.1	(2, 1, 0, 2, 1)
4.1	(0, 2, 1, 2, 1)
4.0	(2, 4, 1, 2, 1)
4.0	(2, 2, 1, 2, 1)
4.0	(2, 1, 2, 4, 1)
4.0	(2, 1, 2, 2, 1)
4.0	(2, 1, 2, 1, 2)
3.9	(3, 2, 1, 2, 1)


So the maximum sum of rewards is 4.1, and there are four ways to get there.